src/cryptonote_core/blockchain.cpp

```diff
@@ -1528,7 +1528,12 @@ bool Blockchain::create_block_template(block& b, const crypto::hash *from_block,
     }
     else
     {
-      if (cur_tx.fee != cur_tx.tx.rct_signatures.txnFee)
+      bool fee_good;
```

Comment:
I don't think this is in the right place, I was confused because I couldn't find where `tx` is defined, here it's probably supposed to be `cur_tx.tx.extra` and then noticed this is in the block macro scope 

```
#if defined(DEBUG_CREATE_BLOCK_TEMPLATE)
...
#endif 
```

`tx_memory_pool::fill_block_template` seems like where this check also needs to go.

This macro is also never activated anywhere, so it might be worth considering deleting as well as it also uses some tx_details which is never used anywhere else except the macro. But I haven't looked carefully into what useful things this macro might enable for debugging.

Reply:
This code is apparently entirely unmaintained.  I tried compiling it with the flag defined and it fails to compile with a reference to `m_tx_pool.m_transactions`, which hasn't existed since Monero 0.10.  And yet people keep adding commits in here...

I'm going to delete it.

Reply:
Oh cool I get to remove a @tewinget FIXME from 2014.

Reply:
> tx_memory_pool::fill_block_template seems like where this check also needs to go.

The check isn't an important one; fill_block_template already does the right thing, this code looks mainly intended to debugging when something wasn't doing the right thing rather than acting as any guard.  (Even if these errors occured it didn't stop anything but just spat out warnings).